File: amqp/src/test/java/edu/upc/ichnaea/amqp/csv/DatasetReaderTest.java
package edu.upc.ichnaea.amqp.csv;

import static org.junit.Assert.*;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import org.junit.Test;

import edu.upc.ichnaea.amqp.data.CsvDatasetReader;
import edu.upc.ichnaea.amqp.model.Dataset;

public class DatasetReaderTest {

    @Test
    public void testFile() throws IOException {
    	InputStream in = getClass().getClassLoader().getResourceAsStream("test.csv");
    	CsvDatasetReader reader = new CsvDatasetReader();
    	reader.setSeparator(';');
    	Dataset dataset = reader.read(new InputStreamReader(in));
    	
    	assertEquals(27, dataset.columnNames().size());
    	
    	dataset.get("RYC2056");
    }
}

File: amqp/php/src/Ichnaea/Amqp/Tests/Xml/DatasetReaderTest.php
<?php

namespace Ichnaea\Amqp\Tests\Model;

use Ichnaea\Amqp\Model\Dataset;
use Ichnaea\Amqp\Xml\DatasetReader;

class DatasetReaderTest extends \PHPUnit_Framework_TestCase
{
    public function testReadingXml()
    {
        $xml = "<dataset>\n";
        $xml .= "<column name=\"test\"><value>1</value><value>2</value><value>3</value></column>\n";
        $xml .= "<column name=\"test2\"><value>3</value><value>4</value></column>\n";
        $xml .= "<column name=\"test3\"><value>5</value><value>6</value><value>7</value></column>\n";
        $xml .= "</dataset>";

        $reader = new DatasetReader();
        $dataset = $reader->read($xml);

        $this->assertEquals(6, $dataset->getColumn("test3")[1], "DatasetReader reads the data correctly.");
    }
}
